Update CHANGELOG for 1.2.1 release (#167)
heremaps/here-olp-sdk-typescript: CHANGELOG.md

## v1.2.1 (05/02/2020)

**Common**

* Updated the development dependencies.

**olp-sdk-dataservice-read**

> * Fix crashes the client if requesting an non-existing tile.

Reviewer: ```suggestion
* Fix crash in `VersionedLayerClient` and `VolatileLayerClient` which happened when requesting a non-existing tile.
```

**olp-sdk-authentication**

> * Fix API break

Reviewer: ```suggestion
* Revert API break in `AuthCredentials`
```

> * Fix crash in `VersionedLayerClient` and `VolatileLayerClient` which happened when requesting a non-existing tile.

Reviewer 2: ```suggestion
* Fixed the crash in `VersionedLayerClient` and `VolatileLayerClient` that happened when a non-existing tile was requested.
```

> * Revert API break in `AuthCredentials`

Reviewer 2: ```suggestion
* Reverted the API break in `AuthCredentials`.
```
